src/events/http.js

```diff
@@ -256,15 +226,31 @@ function httpWrapper(wrappedFunction) {
                             (args[0] instanceof String) || (args[0] instanceof Buffer)
                         )
                     ) {
-                        eventInterface.addToMetadata(
-                            httpEvent, {},
-                            { request_body: args[0].toString() }
-                        );
+                        setJsonPayload(httpEvent, 'request_body', args[0]);
                     }
                     return wrappedWriteFunc.apply(this, args);
                 };
             }
 
+            /**
+             * Wraps 'end' method in a request to terminate the request writing
+             * @param {Function} wrappedEndFunc The wrapped end function
+             * @returns {Function} The wrapped function
+             */
+            function endWrapper(wrappedEndFunc) { // eslint-disable-line no-inner-declarations
+                return function internalEndWrapper(...args) {
+                    if (
+                        (!body || body === '') && args[0] && (
+                            (args[0] instanceof String) || (args[0] instanceof Buffer)
+                        )
+                    ) {
+                        setJsonPayload(httpEvent, 'request_body', args[0]);
+                    }
+                    const result = wrappedEndFunc.apply(this, args);
```

Comment:
you should do this anyway (try catch the above, return this). isn't there a chance you will catch it twice?

Reply:
👍